hedera-mirror-rosetta/app/domain/types/block_test.go

```diff
@@ -0,0 +1,100 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ *
+ * Copyright (C) 2019 - 2020 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+package types
+
+import (
+	"github.com/coinbase/rosetta-sdk-go/types"
+	entityid "github.com/hashgraph/hedera-mirror-node/hedera-mirror-rosetta/app/domain/services/encoding"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func exampleBlock() *Block {
+	return &Block{
+		Index:               2,
+		Hash:                "somehash",
+		ConsensusStartNanos: 10000000,
+		ConsensusEndNanos:   12300000,
+		ParentIndex:         1,
+		ParentHash:          "someparenthash",
+		Transactions: []*Transaction{
+			{
+				Hash: "somehash",
+				Operations: []*Operation{
+					{
+						Index:  1,
+						Type:   "transfer",
+						Status: "pending",
+						Account: &Account{
+							entityid.EntityId{
+								ShardNum:  0,
+								RealmNum:  0,
+								EntityNum: 0,
+							},
+						},
+						Amount: &Amount{Value: int64(400)},
+					},
+				},
+			},
+		},
+	}
+}
+
+func expectedBlock() *types.Block {
+	return &types.Block{
+		BlockIdentifier: &types.BlockIdentifier{
+			Index: 2,
+			Hash:  "0xsomehash",
+		},
+		ParentBlockIdentifier: &types.BlockIdentifier{
+			Index: 1,
+			Hash:  "0xsomeparenthash",
+		},
+		Timestamp: int64(10),
+		Transactions: []*types.Transaction{
+			{
+				TransactionIdentifier: &types.TransactionIdentifier{Hash: "somehash"},
+				Operations:            []*types.Operation{expectedOperation()},
```

Comment:
You could leave `Operations` as an empty array and not use `expectedOperation`. It is tested in the `transactions_test.go` (you have to remove the Operation in the `exampleBlock` as-well)
The important thing to test is that the Transaction, when not empty is transformed into a rosetta one.